cmd/krew/cmd/internal/fetch_tag.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
+const (
+	githubVersionURL = "https://api.github.com/repos/kubernetes-sigs/krew/releases/latest"
+)
+
+// for testing
+var versionURL = githubVersionURL
+
+// FetchLatestTag fetches the tag name of the latest release from GitHub.
+func FetchLatestTag() (string, error) {
+	klog.V(4).Infof("Fetching latest tag from GitHub")
+	response, err := http.Get(versionURL)
+	if err != nil {
+		return "", errors.Wrapf(err, "could not GET the latest release")
+	}
+	defer response.Body.Close()
+
+	var res struct {
+		Tag string `json:"tag_name"`
+	}
+	klog.V(4).Infof("Parsing response from GitHub")
+	if err := json.NewDecoder(response.Body).Decode(&res); err != nil {
+		return "", errors.Wrapf(err, "could not parse the response from GitHub")
+	}
+	klog.V(4).Infof("Fetched latest tag name (%s) from GitHub", res.Tag)
```

Comment:
FYI `http.Get` doesn't handle cases like HTTP 500 , HTTP 429 ...

So you'll get a json error object , which doesn't have tag field, and you'll be doing `return "",nil` is that ok?

Reply:
Yes, that's fine, because this will just look like the request has not finished yet (the notification will only be shown if the tag is not `""`).